Library/SPIClient.m

```diff
@@ -1475,7 +1469,7 @@ - (void)stopPeriodPing {
     SPILog(@"stopPeriodPing");
 
     // If we were already set up, clean up before restarting.
-    [self.pingTimer cancel];
+    _pingTimer = nil;
```

Comment:
noob here. What's the magic behind the timer stopping by simply nullifying the reference?

Reply:
So the retain chain looks like: SPIClient retains SPIRepeatingTimer, which retains the GCD queue+timer, which retain the block, which (hopefully) weakly references SPIClient. If the block strongly referenced SPIClient then we'd have a retain cycle leak. Now ARC magically inserts calls to dispatch_release the GCD queue+timer in SPIRepeatingTimer's dealloc. And when that is called, it cleans everything up, stopping the timer. Since it's not using NSTimer, we don't have to worry about the NSRunLoop retaining everything and getting in the way.

Reply:
wow! magic :)